Content.Server/Fluids/EntitySystems/SpraySystem.cs

```diff
@@ -111,7 +111,7 @@ private void OnAfterInteract(Entity<SprayComponent> entity, ref AfterInteractEve
                 break;
 
             // Spawn the vapor cloud onto the grid/map the user is present on. Offset the start position based on how far the target destination is.
-            var vaporPos = userMapPos.Offset(distance < 1 ? quarter : threeQuarters);
+            var vaporPos = userMapPos.Offset(diff.IsShorterThan(1) ? quarter : threeQuarters);
```

Comment:
This does the length calculation twice (minus sqrt), which is slower than doing it once with the square root. I think you over-estimate how expensive square root is.